Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -165,33 +176,34 @@ private SessionOrCreationToken GetIdleSession(string connStr)
 
         private bool IsAllowedToCreateNewSession()
         {
-            if (!_waitingForSessionToReuseQueue.IsWaitingEnabled())
+            if (!_waitingForIdleSessionQueue.IsWaitingEnabled())
             {
                 s_logger.Debug($"SessionPool - creating of new sessions is not limited");
                 return true;
             }
             var currentSize = GetCurrentPoolSize();
-            if (currentSize < _maxPoolSize)
+            if (currentSize < _poolConfig.MaxPoolSize)
             {
-                s_logger.Debug($"SessionPool - allowed to create a session, current pool size is {currentSize} out of {_maxPoolSize}");
+                s_logger.Debug($"SessionPool - allowed to create a session, current pool size is {currentSize} out of {_poolConfig.MaxPoolSize}");
                 return true;
             }
-            s_logger.Debug($"SessionPool - not allowed to create a session, current pool size is {currentSize} out of {_maxPoolSize}");
+            s_logger.Debug($"SessionPool - not allowed to create a session, current pool size is {currentSize} out of {_poolConfig.MaxPoolSize}");
             return false;
         }
         
         private SFSession WaitForSession(string connStr)
         {
-            var timeout = _waitingForSessionToReuseQueue.GetWaitingTimeoutMillis();
-            s_logger.Info($"SessionPool::WaitForSession for {timeout} ms timeout");
+            if (TimeoutHelper.IsInfinite(_poolConfig.WaitingForSessionIdleTimeout))
+                throw new Exception("WaitingForSessionIdleTimeout cannot be infinite");
```

Comment:
Its inconsistent with what is about waiting timeout in the readme. Check this plz:
https://github.com/snowflakedb/snowflake-connector-net/pull/873/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R170